main.go

```diff
@@ -165,16 +162,27 @@ func exec(logger log.Logger, reg *prometheus.Registry, cfg config) error {
 		}
 	}
 	{
+		{
+			// checks if server is up
+			healthchecks.AddLivenessCheck("http",
+				// TODO: Make use of cfg.server.listen to dynamically configure this too
+				healthcheck.HTTPCheck("http://localhost:8080/", http.MethodGet, http.StatusMovedPermanently, time.Second),
```

Comment:
Do you have an idea on how to extract this address from `cfg.server.listen`? I guess it can always use the localhost as host. We need to switch between https and the port depending on how they are configured. I guess for HTTP we could actually accept any kind of certificate, or do you think we should load the same and verify that it works too?

Reply:
The problem is, that we need to parse something like `:8080` and `url.Parse` actually fails with `parse ":8081": missing protocol scheme`.

Reply:
hmm, if we explicitly create listeners via `l, err := net.Listen("tcp", cfg.server.listen)`, then we can use the `Addr` method of the listener to get the address: https://golang.org/pkg/net/#Listener

This gets us most of the way there, I think. Then we need to look for two special cases `[::]` (IPv6 all addresses) and `0.0.0.0` (IPv4 all addresses) and we can likely sub those both for localhost + the scheme prefix depending on whether certs were given or not